Feature: Screen shake

HIHIQY1 opened the pull request:
Adds screen shake to the project. Includes multiple method overloads for maximum customizability.

Calling a screen shake is simple:

```cs
ScreenShakeController.Instance.StartShake();
```

There are more method overloads for duration, amplitude, and rotations per second.

To add the ScreenShakeController, easiest I've seen is adding an empty GameObject to the project, then moving the active camera to the empty GameObject, then adding the controller script. I've added a demo scene including a demo trigger in `Assets/Scenes/test/screen-shake-example/screen-shake-example.unity`

This PR closes #19

Justwouter commented:
### Notes for #25
- Why is the camera required to be a child of the screenshaker?
You can get a reference to the camera via `[serializefield]` or add a camera as a parameter to either the methods or the `ScreenShakeOptions`

- The singleton is not a singleton per se, there should indeed be a single instance of the class but there is nothing preventing the creation of multiple gameobjects with the `ScreenShakeController` in a scene.

- There is currently no way(?) to have multiple cameras shake without adding them all as children of the shaker.

- Why use var and cast the result when the return types are known (float/double)?

- Have you checked if this works with cinemachine or if cinemachine has a builtin way to add screenshake?

HIHIQY1 commented:
@Justwouter 

To address your notes:
1. This is by design. This allows any other scripts to change the camera's own localTransform
2. fixed
3. That's by design; to make sure all cameras have the same shake applied.
4. fixed, apparently system math stuff returns double
5. I'm not fond of using CineMachine, as I hate not knowing exactly how something works behind the scenes. Inspecting the CineMachine code itself is too much work, and just using our own implementation gives a lot of control, besides knowing exactly what's going on.
